src/Sarif/Visitors/PartitioningVisitor.cs

```diff
@@ -7,6 +7,23 @@
 
 namespace Microsoft.CodeAnalysis.Sarif.Visitors
 {
+
+
+    /// <summary>
+    /// A delegate for a function that returns a value specifying which partition a specified
+    /// result belongs to, or null if the result should be discarded (not placed in any
+    /// partition).
+    /// </summary>
+    /// <param name="result">
+    /// The result to be assigned to a partition.
+    /// </param>
+    /// <typeparam name="T">
+    /// The type of the object returned from the partition function. It must be a reference
+    /// type so that null is a valid value. It must override bool Equals(T other) so that
+    /// two Ts can compare equal even if they are not reference equal.
+    /// </typeparam>
+    public delegate T PartitionFunction<T>(Result result);
```

Comment:
`where T : class, IEquatable<T>`

Reply:
yes, good catch.

---
In reply to: [395180795](https://github.com/microsoft/sarif-sdk/pull/1818#discussion_r395180795) [](ancestors = 395180795)